src/sdk/namespace_fs.js

```diff
@@ -1220,16 +1225,41 @@ class NamespaceFS {
         try {
             const fs_context = this.prepare_fs_context(object_sdk);
             await this._load_bucket(params, fs_context);
-            for (const { key } of params.objects) {
-                // TODO: impl version id and call _find_version_path instead of get_file_path
-                const file_path = this._get_file_path({ key });
-                await this._check_path_in_bucket_boundaries(fs_context, file_path);
-                dbg.log0('NamespaceFS: delete_multiple_objects', file_path);
-                await nb_native().fs.unlink(fs_context, file_path);
-                await this._delete_path_dirs(file_path, fs_context);
+            let res = [];
+            if (this._is_versioning_disabled()) {
+                for (const { key, version } of params.objects) {
+                    if (version) {
+                        res.push({});
+                        continue;
+                    }
+                    try {
+                        const file_path = this._get_file_path({ key });
+                        await this._check_path_in_bucket_boundaries(fs_context, file_path);
+                        dbg.log0('NamespaceFS: delete_multiple_objects', file_path);
+                        await nb_native().fs.unlink(fs_context, file_path);
+                        await this._delete_path_dirs(file_path, fs_context);
+                        res.push({ key });
+                    } catch (err) {
+                        res.push({ err_code: err.code, err_message: err.message });
+                    }
+                }
+            } else {
+                // [{key: a, version: 1}, {key: a, version: 2}, {key:b, version: 1}] => {'a': [1, 2], 'b': [1]}
+                const versions_by_key_map = {};
+                params.objects.forEach(({ key, version_id }) => {
```

Comment:
maybe use for of or try to see if you can use some help function of groupBy or reduce